src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -0,0 +1,514 @@
+        return (String) getCapability(CapabilityType.PLATFORM_NAME);
+    }
+
+    /**
+     * Set the url for the location of the App.
+     *
+     * @param url is a String representing the location of the App
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#APP
+     */
+    public T setApp(String url) {
+        return amend(MobileCapabilityType.APP, url);
+    }
+
+    /**
+     * Set the url for the location of the App.
+     *
+     * @param url is the URL representing the location of the App
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#APP
+     */
+    public T setApp(URL url) {
+        return setApp(url.toString());
+    }
+
+    /**
+     * Get the app location.
+     *
+     * @return String representing app location
+     * @see MobileCapabilityType#APP
+     */
+    public String getApp() {
+        return (String) getCapability(MobileCapabilityType.APP);
+    }
+
+    /**
+     * Set the automation engine to use.
+     *
+     * @param name is the name of the automation engine
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#AUTOMATION_NAME
+     */
+    public T setAutomationName(String name) {
+        return amend(MobileCapabilityType.AUTOMATION_NAME, name);
+    }
+
+    /**
+     * Get the automation engine to use.
+     *
+     * @return String representing the name of the automation engine
+     * @see MobileCapabilityType#AUTOMATION_NAME
+     */
+    public String getAutomationName() {
+        return (String) getCapability(MobileCapabilityType.AUTOMATION_NAME);
+    }
+
+    /**
+     * Set the app to move directly into Webview context.
+     *
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#AUTO_WEBVIEW
+     */
+    public T setAutoWebview() {
+        return setAutoWebview(true);
+    }
+
+    /**
+     * Set whether the app moves directly into Webview context.
+     *
+     * @param bool is whether the app moves directly into Webview context.
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#AUTO_WEBVIEW
+     */
+    public T setAutoWebview(boolean bool) {
+        return amend(MobileCapabilityType.AUTO_WEBVIEW, bool);
+    }
+
+    /**
+     * Get whether the app moves directly into Webview context.
+     *
+     * @return true if app moves directly into Webview context.
+     * @see MobileCapabilityType#AUTO_WEBVIEW
+     */
+    public boolean doesAutoWebview() {
+        return (boolean) getCapability(MobileCapabilityType.AUTO_WEBVIEW);
+    }
+
+    /**
+     * Set the app to clear system files.
+     *
+     * @return this MobileOptions, for chaining.
+     * @see MobileCapabilityType#CLEAR_SYSTEM_FILES
+     */
+    public T setClearSystemFiles() {
+        return setClearSystemFiles(true);
+    }
+
+    /**
+     * Set whether the app clears system files.
+     *
+     * @param bool is whether the app clears system files.
```

Comment:
I think this cap requires better explanation on what `system files` means. These are leftovers from native test frameworks, like logs, debugging data, etc.

Reply:
10726da364ade09fbed223e3a6301459b2055c5a

Reply:
For most of these, additional details can be found by following the `@see` link